src/game_agent.rs

```diff
@@ -0,0 +1,49 @@
+use std::collections::HashMap;
+use super::threading::ThreadMessage;
+use super::primitives;
+use super::utils;
+use super::telegram::{Telegram, Message};
+use primitives::Game;
+
+pub fn new_agent(game_tg_client: Telegram, game_index: usize, playable_games: &Vec<Box<dyn Game>>, reciever: std::sync::mpsc::Receiver<ThreadMessage>) {
+    let game = Box::leak(playable_games[game_index].get_new_instance());
+    std::thread::spawn(move || {
+        let mut message_list: HashMap<i64, i64> = HashMap::new();
+        let client = game_tg_client;
+        let game = game;
+        game.init();
+        let mut game_is_running = true;
+        while game_is_running {
+            let message = reciever.recv().unwrap();
+            let status = match message {
+                ThreadMessage::AddPlayer(p) => vec![game.add_player(p.clone()).unwrap_or_else(|x| primitives::GameStatus::NotifyUser(p, x.to_owned()))],
+                ThreadMessage::Start => vec![game.start(), primitives::GameStatus::NotifyRoom(game.get_status())],
+                ThreadMessage::HandleMove(p, c) => {
+                    let mut tmp = game.handle_move(&p, c);
```

Comment:
tmp????

Reply:
Temporary variable used to dispatch the game status alongside the game originated messages